homeassistant/components/modbus/sensor.py

```diff
@@ -212,7 +70,7 @@ def __init__(
         self._scale = entry[CONF_SCALE]
         self._offset = entry[CONF_OFFSET]
         self._precision = entry[CONF_PRECISION]
-        self._structure = structure
+        self._structure = entry[CONF_STRUCTURE]
```

Comment:
```suggestion
        self._structure = entry.get[CONF_STRUCTURE]
```
Many configs do not have CONF_STRUCTURE, and it should not be demanded (even if the validators adds it).

Reply:
👍🏼